lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -231,11 +233,14 @@ applyBlock !b (Wallet !u !pending _ s bh) =
 --
 applyBlocks
     :: forall s t. (DefineTx t)
-    => [Block (Tx t)]
+    => NonEmpty (Block (Tx t))
     -> Wallet s t
-    -> [(Map (Hash "Tx") (Tx t, TxMeta), Wallet s t)]
+    -> NonEmpty (Map (Hash "Tx") (Tx t, TxMeta), Wallet s t)
 applyBlocks blocks cp0 =
-    NE.tail $ NE.scanl (flip applyBlock . snd) (mempty, cp0) blocks
+    -- NOTE:
+    -- The list of blocks is of length `n > 0`, scanl generates a list of length
+    -- `n+1`, so its tail is necessarily of length `n > 0`!
+    NE.fromList $ NE.tail $ NE.scanl (flip applyBlock . snd) (mempty, cp0) blocks
```

Comment:
We can avoid the partial function (and comment) here by applying the first block separately:
```hs
applyBlocks (block :| blocks) cp0 =                                                                                                                                                   
    NE.scanl (flip applyBlock . snd) (applyBlock block cp0) blocks      
```

Reply:
Fair point :+1:

Reply:
Change done!